ARCHITECTURE.md

```diff
@@ -0,0 +1,241 @@
+[DPoP tokens](https://tools.ietf.org/html/draft-ietf-oauth-dpop-01) and 
+[Dynamic Client Registration](https://openid.net/specs/openid-connect-registration-1_0.html),
+which are implemented in `@inrupt/oidc-client-ext`.
+
+The four modules are available in the [packages directory](./packages).
+
+## OAuth2.0/OpenID Connect
+
+This library aims at helping developers authenticating users to their applications
+using the [OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html)
+protocol (often abbreviated OIDC). OIDC is a protocol based on the
+[OAuth2.0](https://tools.ietf.org/html/rfc6749) framework. In order to understand
+how the library internally works, some understanding of OAuth/OIDC is preferable.
+
+### Helpful resources
+
+Here is a list of things that helped this library's developers get a better
+understanding of OAuth/OIDC: 
+- [OAuth 2 in Action](https://www.manning.com/books/oauth-2-in-action), by Justin Richer and Antonio Sanso
+- [OAuth masterclass](https://www.youtube.com/watch?v=egfyV2NV9Mw), by Justin Richer
+- [How To Securely Implement Authentication in Single Page Applications](https://betterprogramming.pub/how-to-securely-implement-authentication-in-single-page-applications-670534da746f), by Dennis Stötzel
+
+### Solid-OIDC
+
+Solid extends the OIDC protocol in order to make it fit into a decentralized ecosystem. 
+The [Solid-OIDC specification](https://solid.github.io/authentication-panel/solid-oidc/) 
+is the reference document to learn about this extension.
+
+In particular, Solid-OIDC introduces the notion of Client WebID, which enables
+Client-managed identifiers, and prevents from requiring Clients to get Issuer-managed
+identifiers through either static or dynamic client registration. Solid-OIDC also
+makes mandatory the support for [Key-bound Access Tokens](https://tools.ietf.org/html/draft-fett-oauth-dpop-04)
+(referred to as DPoP tokens), which is only optional in traditional OIDC, where
+Bearer tokens are the default option. DPoP tokens cannot be replayed by a Resource
+Server to another Resource Server, which is an important security feature in a
+decentralized ecosystem such as Solid's.
+
+### A short glossary
+
+Here is a list of terms having a specific meaning in the context of OIDC:
+- **Resource Server**: The server hosting private resources. In our case, a Solid
+server. A Resource Server receives requests authenticated with an Access Token.
+Example: https://pod.inrupt.com. 
+- **Resource Owner**: the user, who owns some private resources 
+- **OIDC issuer**: the Solid Identity Provider, which issues Access Tokens, ID
+tokens, and Refresh Tokens. These tokens tell the Resource Server that the user
+has control over a certain identity (WebID), which can then use that information
+to decide whether to give or deny access. Example: https://broker.pod.inrupt.com.
+- **Client**: the application the Resource Owner uses to access its resources on
```

Comment:
The notion of Resource Owner in the OIDC sense is broader than the notion of Pod Owner in Solid: anyone with a WebID is a Resource Owner, in the authentication sense. Whether a particular Resource Owner is authorized to access a particular Resource is a different matter, up to the authozization system, so ACP or ACL, which is outside of the scope of this document. Do you think that's a helpful precision to add ?

Reply:
Yes please.  Because the Resource Owner definition follows that of the Resource Server - which ends with example https://pod.inrupt.com, it because easier to make the mental jump to equate Resource Owner to Pod Owner.  So, explicitly stating that it's anyone with a WebID clarification would definitely help.